pkg/handler/handler.go

```diff
@@ -117,8 +117,11 @@ func WebhookHandler(w http.ResponseWriter, r *http.Request, reconciler gitopscon
 	log.Info("webhook handling concluded correctly")
 }
 
-func repoURLMatch(instance *gitopsv1alpha1.GitOpsConfig, event *github.PushEvent) bool {
-	return strings.Contains(instance.Spec.TemplateSource.URI, *event.Repo.FullName) || strings.Contains(instance.Spec.ParameterSource.URI, *event.Repo.FullName)
+func repoURLAndRefMatch(instance *gitopsv1alpha1.GitOpsConfig, event *github.PushEvent) bool {
```

Comment:
Please add a unit test for this func, with at least 1 positive and 1 negative test case. Some helpful links for learning how to write unit tests in Go in a standard way:

- https://golang.org/doc/code.html#Testing
- https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Suggested initial template for the test function:

```
func TestRepoURLAndRefMatch(t *testing.T) {
    tests := []struct{
        comment  string
        instance *gitopsv1alpha1.GitOpsConfig
        event    *github.PushEvent
        want     bool
    }{
        {
             comment: "full match",
             ...
             want: true,
        },
        {
             comment: "parameterSource.Ref differs",
             ...
             want: false,
        },
        {
             comment: "parameterSource.URI differs",
             ...
             want: false,
        },
    }

    for _, tt := range tests {
         result := repoURLAndRefMatch(tt.instance, tt.event)
         if result != tt.want {
             t.Errorf("%q: expected %v, got %v", tt.comment, tt.want, match)
         }
    }
}
```